### PR TITLE
[ALLUXIO-3167] Allow application level TTL

### PR DESCRIPTION
**Pull request by Reidddddd**
https://alluxio.atlassian.net/browse/ALLUXIO-3167

### CODE REVIEW

File: core/server/master/src/main/java/alluxio/master/file/InodeTtlChecker.java [hunk 1]
```diff
@@ -71,16 +71,17 @@ public void heartbeat() {
...
+            LOG.info("Path {} TTL has expired, performing action {}", path.getPath(), ttlAction);
             switch (ttlAction) {
               case FREE:
                 // public free method will lock the path, and check WRITE permission required at
                 // parent of file
                 if (inode.isDirectory()) {
-                  mFileSystemMaster
-                      .free(path, FreeOptions.defaults().setForced(true).setRecursive(true));
+                  mFileSystemMaster.delete(path,
+                      DeleteOptions.defaults().setRecursive(true).setAlluxioOnly(true));
```
Comment (Reidddddd) [inline core/server/master/src/main/java/alluxio/master/file/InodeTtlChecker.java]: Temporarily solution. I filed a jira to describe it, for more information please see [ALLUXIO-3182](https://alluxio.atlassian.net/browse/ALLUXIO-3182)

**Comment by Reidddddd** [comment]
Let's start discussions before fixing those failed unit tests.

**Comment by aaudiber** [comment]
@Reidddddd Could you write up a design doc for the new TTL functionality and share it on the JIRA ticket? I've left some thoughts on the ticket.

**Comment by Reidddddd** [comment]
I might make some mistakes on updating pull request...
sorry..
new pr is https://github.com/Alluxio/alluxio/pull/7317

**Comment by Reidddddd** [comment]
Close this one, marked as duplicate.

**Pull request closed without merging**